### ISSUE TITLE
expo-local.authentication, stucks in app_cancel

### ISSUE THREAD

**Issue by bdomonkos** [expo/expo#12614]
### Summary

Hi 
I am a react-native  developer at Blackrock. We have to integrate the biometric authentication to our application.

The issue: When a user puts the application into background during the biometric/pattern..etc.. authentication. For example if the lock screen is pattern and its visible and the user puts the app into background and bring back to foreground. 

On some devices(google pixel), the callback of the promise won't be invoked after putting the device to background. And the localauthenticaiton stucks in app_cancel status and the user wont be able to use the biometric authentication, until an application restart.

We tried to use the cancelAuthenticate but it did not work.

Any help could be appreciated!

Thanks
Domonkos

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

uni modules

### Environment

---

### Reproducible demo or steps to reproduce from a blank project

1.react-native init 
2. add uni-modules
3. add expo-local-auth
4. start an authentication(authenticateAsync)
5. put the app to background
6. bring back to foreground
7. try to start on more bio auth it should fail with app_cancel

**Comment by AdamJNavarro** [comment]
Hey @bdomonkos, could you please create a public repo with a simple implementation/reproduction of this issue? Being able to quickly clone and test this helps greatly in our debugging efforts and expedites the support process!

Cheers,
Adam

**Comment by bdomonkos** [comment]
https://github.com/bdomonkos/bioAuthTest

**Comment by bdomonkos** [comment]
I did some research. When upgrade the biometrics dependency to androidx.biometric:biometric:1.1.0 the issue was solved. But some new problem appeared. I had to catch the result of the DeviceCredentialHandlerActivity and I had to do some modification but its working.

Are there any reason that the expo-local-authentication does not use the latest biometrics library?

Btw the fallback scenario still not working with fingerprint. The Android biometrics library always invoke a dismiss after a fingerprint scenario, but after the fallback this step is missing and it leads to problems.
